File: app/src/main/java/com/apps/LauncherRecentAdapter.java
package com.apps;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.yuki.yukihub.R;

import java.util.ArrayList;
import java.util.List;

public class LauncherRecentAdapter extends RecyclerView.Adapter<LauncherRecentAdapter.Holder> {
    private final List<LauncherRepository.RecentItem> items = new ArrayList<>();

    public void submit(List<LauncherRepository.RecentItem> newItems) {
        items.clear();
        if (newItems != null) items.addAll(newItems);
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_launcher_recent, parent, false);
        return new Holder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        LauncherRepository.RecentItem item = items.get(position);
        holder.icon.setText(item.iconText);
        holder.title.setText(item.title);
        holder.meta.setText(item.timeAndDuration);
        holder.status.setText(item.status);
        LauncherTheme.textPrimary(holder.icon);
        LauncherTheme.textPrimary(holder.status);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    static final class Holder extends RecyclerView.ViewHolder {
        final TextView icon;
        final TextView title;
        final TextView meta;
        final TextView status;

        Holder(@NonNull View itemView) {
            super(itemView);
            icon = itemView.findViewById(R.id.recentIcon);
            title = itemView.findViewById(R.id.recentTitle);
            meta = itemView.findViewById(R.id.recentMeta);
            status = itemView.findViewById(R.id.recentStatus);
        }
    }
}

File: app/src/main/java/com/apps/game/LauncherRecentAdapter.kt
package com.apps.game

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.apps.data.LauncherRepository
import com.apps.theme.LauncherTheme
import com.apps.widget.LauncherTabletPortraitScaler
import com.yuki.yukihub.R

class LauncherRecentAdapter : RecyclerView.Adapter<LauncherRecentAdapter.Holder>() {
    private val items = mutableListOf<LauncherRepository.RecentItem>()

    fun submit(newItems: List<LauncherRepository.RecentItem>?) {
        items.clear()
        newItems?.let(items::addAll)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): Holder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_launcher_recent, parent, false)
        LauncherTabletPortraitScaler.apply(view)
        return Holder(view)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        val item = items[position]
        holder.icon.text = item.iconText
        holder.title.text = item.title
        holder.meta.text = item.timeAndDuration
        holder.status.text = item.status
        LauncherTheme.textPrimary(holder.icon)
        LauncherTheme.textPrimary(holder.status)
    }

    override fun getItemCount(): Int = items.size

    class Holder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val icon: TextView = itemView.findViewById(R.id.recentIcon)
        val title: TextView = itemView.findViewById(R.id.recentTitle)
        val meta: TextView = itemView.findViewById(R.id.recentMeta)
        val status: TextView = itemView.findViewById(R.id.recentStatus)
    }
}
